elastic/kibana
# [Chart expressions] new metric vis expression

## Summary

Prepares for https://github.com/elastic/kibana/issues/134242

Introduces a new metric vis chart expression function + renderer that uses the new @elastic/elastic-charts metric. Also, renames the existing metric vis to reflect its legacy status.

In other words, `src/plugins/chart_expressions/expression_metric` is now the new function and the existing one has moved to ` src/plugins/chart_expressions/expression_legacy_metric`.  Unfortunately, this makes the file count high and diff a bit hard-to-follow. The meat of the new logic is actually mostly in this [one file](https://github.com/elastic/kibana/pull/135461/files#diff-09b11c292a65e3dbf5dc4614bc9b3f82e64143966a608c16c1d4dd51e7ae363e).

### Testing
Manual testing can be performed on a Canvas workpad.

To get started,
- add `kibana_sample_data_ecommerce`
- apply the currency field formatter to `products.base_price` and `products.min_price` in the data view
- paste the following expression into a new Canvas shape using the expression editor
```
kibana
| esaggs index={indexPatternLoad id="ff959d40-b880-11e8-a6d9-e546fe2bba5f"} 
  aggs={aggTerms id="0" enabled=true schema="segment" field="day_of_week" orderBy="1" order="desc" size=6 otherBucket=false otherBucketLabel="Other" missingBucket=false missingBucketLabel="(missing value)"}
  aggs={aggMedian id="1" enabled=true schema="metric" field="products.base_price"} metricsAtAllLevels=false partialRows=false
  aggs={aggMedian id="2" enabled=true schema="metric" field="products.min_price"} metricsAtAllLevels=false partialRows=false
| metricVis metric="col-2-2" secondaryMetric="col-1-1" breakdownBy="col-0-0" progressMax="col-1-1" maxCols=3 palette={palette color="purple" color="orange" stop=13.5 stop=20 reverse=false range="number" rangeMax=30 rangeMin=0}
```

Formatting is tied to the field formatters, so you can try out currency, bytes, percentage, duration, or just vanilla numbers. Just keep in mind, that this vis doesn't respect user-defined formatting patterns.

You can see more about expected formatting behavior in the [unit test](https://github.com/elastic/kibana/pull/135461/files#diff-e7cd6d27a330121568d3ea2f4149be9cc4019f8797058b6a3175d10812f3d63cR714) and in the original issue.

### Screenshots
<img width="541" alt="Screen Shot 2022-06-30 at 10 12 51 AM" src="https://user-images.githubusercontent.com/315764/176718391-747e7525-6284-40f9-bcbe-ff1d58195663.png">
<img width="536" alt="Screen Shot 2022-06-30 at 10 15 35 AM" src="https://user-images.githubusercontent.com/315764/176718394-4b5f2c34-c392-4c46-8284-d240527b55a0.png">


### Checklist

Delete any items that are not applicable to this PR.
